extensions/wikia/Chat2/ChatController.class.php

```diff
@@ -11,6 +11,12 @@ public function executeIndex() {
 
 		Chat::info( __METHOD__ . ': Method called' );
 
+		// SUS-934: Abort early if someone is trying to connect directly without permission
+		if ( !Chat::canChat( $this->wg->User ) ) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)